python/ray/serve/http_proxy.py

```diff
@@ -44,40 +107,38 @@ def __init__(self, controller_name):
             description="The number of HTTP requests processed.",
             tag_keys=("route", ))
 
-    async def setup(self):
-        await self.router.setup_in_async_loop()
-
     async def _update_route_table(self, route_table):
         logger.debug(f"HTTP Proxy: Get updated route table: {route_table}.")
         self.route_table = route_table
 
-    async def receive_http_body(self, scope, receive, send):
-        body_buffer = []
-        more_body = True
-        while more_body:
-            message = await receive()
-            assert message["type"] == "http.request"
+        routes = [
```

Comment:
this is recreating the routes each time a new routes is added right? is there any overhead?

Reply:
That's right.  The Route constructor method is pretty simple, so I don't think there will be much overhead.